### PR TITLE
27 update the tests to work on a specific collection

### LINKED ISSUE #27: Update the tests to work on a specific collection
**Is your feature request related to a problem? Please describe.**
We can't easily launch our backend tests

**Describe the solution you'd like**
A setup function to create a specific collection for tests purposes 

- [ ] Add a new function to change the default collection value
- [ ] Modified tests to work on a specific collection

### PR DESCRIPTION
**Pull request by altristid**
## Issue ticket number
Closes #

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have written unit tests for my code if applicable

### CODE REVIEW

File: back/services/arangoManager.py [hunk 1]
```diff
@@ -6,16 +6,18 @@
 
 WORKS_COLLECTION_NAME = "works"
 
-
+#setup handle a variable default_collection with a default value at work
+#this function can create now handle a new collection by default if this parameter is define
 def setup():
-    global db
+    global db,WORKS_COLLECTION_NAME
```
Comment (Tomansion) [inline back/services/arangoManager.py]: Useless Change ?

File: back/services/arangoManager.py [hunk 2]
```diff
@@ -6,16 +6,18 @@
...
     # Load config
     config = get_config()
     HOST = config["ARANGODB"]["HOST"]
     PORT = config["ARANGODB"]["PORT"]
     DATABASE = config["ARANGODB"]["DATABASE"]
     USER = config["ARANGODB"]["USER"]
     PASSWORD = config["ARANGODB"]["PASSWORD"]
+    # Define WORKS_COLLECTION_NAME
```
Comment (Tomansion) [inline back/services/arangoManager.py]: ?

File: back/services/arangoManager.py [hunk 3]
```diff
@@ -7,15 +7,18 @@
 WORKS_COLLECTION_NAME = "works"
 
 
+# setup handle a variable default_collection with a default value at work
+# this function can create now handle a new collection by default if this parameter is define
 def setup():
-    global db
+    global db, WORKS_COLLECTION_NAME
```
Comment (Tomansion) [inline back/services/arangoManager.py]: Why ?

File: back/tests/test_arango_manager.py [hunk 4]
```diff
@@ -10,23 +10,28 @@
...
 
 
-@pytest.fixture(scope="session", autouse=True)
+# The scope module is define to use the setup database function for all tests
+@pytest.fixture(scope="module", autouse=True)
 def setup_database():
     # This code will run before the first test
     setup()
+    # Define a specific collection to isolate our tests from the default collection
+    setup_collection("test")
```
Comment (Tomansion) [inline back/tests/test_arango_manager.py]: "test" will be the test collection that stores Artworks ? If yes, rename it more clearly.
`artwork_test` ?

File: back/tests/test_arango_manager.py [hunk 5]
```diff
@@ -10,23 +10,28 @@
...
 test_doc_id_1 = None
 test_doc_id_2 = None
 test_doc_id_3 = None
 original_number_of_works = None
+test_collection = "test"
 
 
-@pytest.fixture(scope="session", autouse=True)
+# The scope module is define to use the setup database function for all tests
+@pytest.fixture(scope="module", autouse=True)
```
Comment (Tomansion) [inline back/tests/test_arango_manager.py:25]: Why was this change required ?

File: back/services/arangoManager.py [hunk 6]
```diff
@@ -33,6 +36,11 @@ def setup():
     print(" - Connection established")
 
 
+def setup_collection(collection_name):
```
Comment (Tomansion) [inline back/services/arangoManager.py]: What do you think about `setup_collection_name` ? Without that, we may think that this function changes something in arrango.

Comment (Tomansion) [inline back/services/arangoManager.py]: Setup witch collection ? Their will be more than one collection in the future

File: back/services/arangoManager.py
Comment (Tomansion) [inline back/services/arangoManager.py:1]: Please complete the description of the PR with what changed

Comment (Tomansion) [inline back/services/arangoManager.py:1]: It is best that the pipeline pass before requesting review

**Review by Tomansion — CHANGES_REQUESTED** [review]
Few changes required

File: back/services/arangoManager.py (hunk 6 above)
Comment (altristid) [reply back/services/arangoManager.py]: Should I change it for setup_works_collection_name since we will have more than one collection in the future ?

File: back/tests/test_arango_manager.py (hunk 5 above)
Comment (altristid) [reply back/tests/test_arango_manager.py:25]: I changed it because I identified the session scope as a mistake, because we weren't able to request the arangodatabase with our setup, now the setup procedure  is applied before any test in the test_arango_manager file.

**Review by Tomansion — APPROVED** [review]
Good apart from the missing PR description
